### ISSUE TITLE
Test for quotation marks etc.

### ISSUE THREAD

**Issue by sean-wood** [FreeUKGen/MyopicVicar#846]

**Comment by DeniseColbert** [comment]
How can I help with this?

**Comment by Captainkirkdawson** [comment]
Not sure what the issue is about

**Comment by benwbrum** [comment]
I believe that this is a task to address "smart quotes" -- the unicode
characters that some software packages insert whenever a user types ' or
".  We should verify that these are actually a problem, then eliminate them.

Ben

**Comment by AlOneill** [comment]
Or rather **replace** them with their plain equivalents. In the searchable fields -- surnames and forenames.

An example would be surnames such as O'Neill, with apostrophes/single quotes: the smart quote version isn't found in a search, *as far as I can tell*.

**Comment by benwbrum** [comment]
Thanks for clarifying!  I'd like to take this one, since I've done very
similar work on other projects.

**Comment by PatReynolds** [comment]
Trying to locate the other story about this issue ... but I believe it is peculiar to macs.  Executive think that we should be able to handle unicode, but I'm not sure if the problem is handling unicode, or something which macs do ....

**Comment by Captainkirkdawson** [comment]
We do handle unicode if the file is presented as unicode. If they tell us it is some other character set then we have issues as would any system.

Not sure the "other" story ever became a story. It involved the auto conversion of some edits and spreadsheets (Sean ad Alison were documenting)

**Comment by Sherlock21** [comment]
No,Pat. its not unique to Macs. its evident there because maybe the Mac people all use spreadsheets whereas there is the option on windows. Be assured its a common feature of Windows, Linux and Mac. - 

Its a matter of turning off the Auto Correct feature on the Spreadsheet Application that is being used, have the latest revision of the operating system of the Computer and the Spreadsheet,  AND put the right Character set code in the file header.

**Comment by edickens** [comment]
If I remember correctly, this started when files "Barfed" the server because of illegal characters created by MAC computers, and these were quotation marks.  A solution has been put in the Help.  However, I have not see any of the Barfing messages recently.
But it could also apply to the attached message.  Here quotation marks are put in by some versions of Excel and this causes problems.


![capture](https://cloud.githubusercontent.com/assets/8876165/17560279/63394c3e-5f19-11e6-99f1-adef2cc52410.JPG)

**Comment by Captainkirkdawson** [comment]
The application uses unicode for storing information. Unicode permits the use of "smart quotes.

The issue is can users submit smart characters in their files that are then incorporated into the fields and appropriate records are retrieved by the search engine.

Are there limitations on how such files created and submitted.

Can one submit O'Neil as a name and have it retrieved by the system

There is an excellent page that describes Smart Quotes for Smart People

http://smartquotesforsmartpeople.com/

"Don't be dumb"  “You’re smart!”  Good typography uses smart quotes, not dumb quotes

“Smart quotes,” the correct quotation marks and apostrophes, are curly or sloped. "Dumb quotes," or straight quotes, are a vestigial constraint from typewriters when using one key for two different marks helped save space on a keyboard. Unfortunately, many improper marks make their way onto websites because of dumb defaults in applications and CMSs. Luckily, using correct quotation marks and apostrophes today is easier than you think. The simplest method is to use charset utf-8 on your webpages and just type or copy and paste the actual punctuation marks. (See link for codes)

**Comment by PatReynolds** [comment]
There seems to be a difference between the search results for O'Neil marriages in Lancashire with a 'proper' quote (30) and with a 'smart' quote (0).  I wouldn't want to stop people using unicode to search - rather I think we need a way of taking the unicode characters entered and looking them up as if they were something else (e.g. the Dutch ij ligature is looked up as ij, the letter 'eth/edd' is looked up as th), and vice versa.  Then it wouldn't matter if someone twys Bijl (four letters, because that's how the British vicar wrote it) but someone searched for Bĳl (three letters) because that's how they write the family name today.  Which doesn't get me any further on giving this a milestone.  So I'll pick 1.4, and we can revise if needed.

**Comment by AlOneill** [comment]
Ali Hill has investigated smart quotes and **forename abbreviations** on test3. I can confirm that forename abbreviations with *plain quotes* are returned on a search for the full forename, whereas abbreviations with *smart quotes* are **not** found.

In order to find forenames abbreviations containing *smart quotes*, one has to search explicitly, e.g.   w’m -- but results containing plain quotes are  **not** returned for this search.

**Comment by AlOneill** [comment]
In response to Pat's comments, should we consider treating i and j as equivalent? What about u and v?

As an example, in transcribing from a transcript, I have come across the forename Jsobel .
